DynamicProgramming/CoinChange.java

```diff
@@ -23,14 +25,15 @@ public static void main(String[] args) {
      * @param amount The amount for which we need to find the change
      *               Finds the number of combinations of change
      **/
-    public static int change(int[] coins, int amount) {
+    @SuppressWarnings("cast.unsafe")
+    public static int change(@Positive int[] coins,@NonNegative int amount) {
 
-        int[] combinations = new int[amount + 1];
-        combinations[0] = 1;
+        int[] combinations =(int @MinLen(1) []) new int[amount + 1];
+        combinations[0] = 1; /* This Line gives cast.unsafe warning because the compiler is unable to statically verify that the length of the array "combinations" is greater than 1*/
 
-        for (int coin : coins) {
+        for (@Positive int coin : coins) {
             for (int i = coin; i < amount + 1; i++) {
-                combinations[i] += combinations[i - coin];
+                combinations[i] += combinations[(@Positive int)(i - coin)];
```

Comment:
I don't think this cast is safe. In the first iteration of this loop, `i` = `coin`, so `i - coin` will be zero, which is not positive.

Reply:
I made this mistake unknowingly. I have changed it to @NonNegative.